### PR TITLE
Add MariaDB user management

### PR DESCRIPTION
**Pull request by Kidswiss**
## Summary

* Added MariaDB usermanagement analog to PostgreSQL usermanagement
## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.
- [ ] Link this PR to related issues.

### CODE REVIEW

File: pkg/comp-functions/functions/vshnmariadb/user_management.go [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+func addConnectionDetail(comp common.Composite, svc *runtime.ServiceRuntime, secretName, username, dbname string, connectionDetailRef *xpv1.SecretReference) {
+	userpassCD, err := svc.GetObservedComposedResourceConnectionDetails(secretName)
+	if err != nil {
+		svc.Log.Error(err, "cannot get userpassword from secret")
+		svc.AddResult(runtime.NewWarningResult(fmt.Sprintf("cannot get userpassword from secret: %s", err)))
+	}
+
+	compositeCD := svc.GetConnectionDetails()
+
+	// url := getPostgresURLCustomUser(compositeCD, string(compositeCD["MARIADB_HOST"]), username)
```
Comment (wejdross) [inline pkg/comp-functions/functions/vshnmariadb/user_management.go:95]: leftover?

File: pkg/comp-functions/functions/vshnmariadb/user_management.go [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+			TLS: &tls,
+			Credentials: my1alpha1.ProviderCredentials{
+				Source: "MySQLConnectionSecret",
+				ConnectionSecretRef: &xpv1.SecretReference{
+					Name:      "provider-conf-credentials",
+					Namespace: comp.GetInstanceNamespace(),
+				},
+			},
+		},
+	}
```
Comment (wejdross) [inline pkg/comp-functions/functions/vshnmariadb/user_management.go:174]: I assume it requires specific field names and this is why we don't reuse existing secrets?

File: pkg/comp-functions/functions/vshnmariadb/user_management.go [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+	if err != nil {
+		svc.AddResult(runtime.NewWarningResult(fmt.Sprintf("cannot apply the provider config for provider sql: %s", err)))
+		svc.Log.Error(err, "cannot apply the provider config for provider sql")
+	}
+}
+
+// We check if the database is already specified.
+// If not it will be added.
+// This should handle cases where there are mutliple users pointing to the same
+// database, and one is deleted, that the database is not dropped.
```
Comment (wejdross) [inline pkg/comp-functions/functions/vshnmariadb/user_management.go:186]: how would You like to delete user or database? Fields in Claim stays forever and we do not have a switch like enabled/prune or anything like that
```
type VSHNAccess struct {
	// User specifies the username. If all other fields are left empty
	// then a new database with the same name and all permissions will be created.
	// +kubebuilder:validation:Required
	User *string `json:"user"`

	// Database is the name of the database to create, defaults to user.
	Database *string `json:"database,omitempty"`

	// Privileges specifies the privileges to grant the user. Please check
	// the database's docs for available privileges.
	Privileges []string `json:"privileges,omitempty"`

	// WriteConnectionSecretToReference specifies the namespace and name of a
	// Secret to which any connection details for this user should
	// be written.
	// If not specified, a secret with the name $claimname-$username will be
	// created in the namespace where the claim is located.
	WriteConnectionSecretToReference *xpv1.SecretReference `json:"writeConnectionSecretToRef,omitempty"`
}
```

Comment (wejdross) [reply pkg/comp-functions/functions/vshnmariadb/user_management.go:186]: ok, I saw second MR (docs) and now I get how deletion works :D

File: pkg/comp-functions/functions/vshnmariadb/user_management.go (hunk 2 above)
Comment (Kidswiss) [reply pkg/comp-functions/functions/vshnmariadb/user_management.go:174]: Yes, the secret has to look exactly like this.

File: pkg/comp-functions/functions/vshnmariadb/user_management.go (hunk 1 above)
Comment (Kidswiss) [reply pkg/comp-functions/functions/vshnmariadb/user_management.go:95]: yes

**Review by wejdross — APPROVED** [review]

**Review by zugao — APPROVED** [review]
LGTM

**Pull request merged**